src/common/transformations/src/transformations/common_optimizations/reduce_reshape_fusion.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/reduce_reshape_fusion.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "itt.hpp"
+#include "openvino/op/util/reduction_base.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::ReduceReshapeFusion::ReduceReshapeFusion() {
+    MATCHER_SCOPE(ReduceReshapeFusion);
+
+    const auto reduce_axes = pattern::wrap_type<opset9::Constant>();
+    const auto reduce = pattern::wrap_type<op::util::ArithmeticReductionKeepDims, op::util::LogicalReductionKeepDims>(
+        {pattern::any_input(), reduce_axes},
+        pattern::has_static_shape());
+    const auto reshape =
+        pattern::wrap_type<opset9::Reshape>({reduce, pattern::any_input()}, pattern::has_static_shape());
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        auto reshape_node = pattern_map.at(reshape).get_node_shared_ptr();
+        const auto reduce_node = pattern_map.at(reduce).get_node_shared_ptr();
+        const bool keep_dims =
```

Comment:
is it possible to move `get_keep_dims()` to `ReductionBase` class?

Reply:
We can, but it means extending API of `ArithmeticReduction` and `LogicalReduction` with `get_keep_dims` while these classes don't take `keep_dims` argument. However, we can return false for them. Is it ok for you?

Reply:
I believe we can extend `ReductionBase` with `virtual bool get_keep_dims() const { return false; }`
@jane-intel do you have any objections?

Reply:
I've pushed it to check if any problems but it looks good for me

Reply:
Maybe introduce interface class `IKeepDims` and add it to `ArithmeticReductionKeepDims` , `LogicalReductionKeepDims`?
Then you can cast to this interface class and use interface methods.